vms/platformvm/block/executor/proposal_block_test.go

```diff
@@ -392,57 +392,46 @@ func TestBanffProposalBlockUpdateStakers(t *testing.T) {
 	// Staker5:                                     |--------------------|
 
 	// Staker0 it's here just to allow to issue a proposal block with the chosen endTime.
-	staker0RewardAddress := ids.ShortID{2}
 	staker0 := staker{
-		nodeID:        ids.NodeID(staker0RewardAddress),
-		rewardAddress: staker0RewardAddress,
+		nodeID:        ids.BuildTestNodeID([]byte{0xf0}),
```

Comment:
This is a nasty test where multiple stakers may join and leave the staker set. The order in which they do it is asserted (so wrong order will make test fail) but it may depend on the staker.TxID, which in turns depend on *every* feature of the transaction creating the staker.
Here I avoid using ids.GenerateTestNodeID because it wil give different result depending on the order in which UTs are executed. Hard-coding nodeID as I did here should make the test more robust

Reply:
^ Probably worth including this in a code comment